src/PIL/Image.py

```diff
@@ -1341,7 +1347,9 @@ def getbbox(self, *, alpha_only: bool = True) -> tuple[int, int, int, int]:
         self.load()
         return self.im.getbbox(alpha_only)
 
-    def getcolors(self, maxcolors=256):
+    def getcolors(
+        self, maxcolors: int = 256
+    ) -> list[tuple[int, int | tuple[int, ...]]] | None:
```

Comment:
Let's deal with this after this release:
```
mypy: commands[0]> mypy src
src/PIL/Image.py:1372: error: Incompatible return value type (got
"List[Tuple[Any, int]]", expected
"Optional[List[Tuple[int, Union[int, Tuple[int, ...]]]]]")  [return-value]
                return out
                       ^~~
Found 1 error in 1 file (checked 102 source files)
mypy: exit 1 ([16](https://github.com/python-pillow/Pillow/actions/runs/8504209278/job/23290792512?pr=7924#step:8:17).84 seconds) /home/runner/work/Pillow/Pillow> mypy src pid=2141
```


```suggestion
    def getcolors(self, maxcolors: int = 256):
```